federation/src/config/index.ts

```diff
@@ -73,8 +73,10 @@ if (
 const federation = {
   // FEDERATION_DHT_TOPIC: process.env.FEDERATION_DHT_TOPIC || null,
   // FEDERATION_DHT_SEED: process.env.FEDERATION_DHT_SEED || null,
-  FEDERATION_PORT: process.env.FEDERATION_PORT || 5010,
+  //  FEDERATION_COMMUNITY_API_PORT:
+  //    process.env.FEDERATION_COMMUNITY_API_PORT || 5000,
   FEDERATION_API: process.env.FEDERATION_API || '1_0',
+  FEDERATION_PORT: process.env.FEDERATION_PORT || 5010,
   FEDERATION_COMMUNITY_URL: process.env.FEDERATION_COMMUNITY_URL || null,
```

Comment:
We should consider to pass the full URL, the service is serving on, in this config variable - this would allow us to define various endpoints

`myurl.com/api/10/`
`myurl.com/api/1_1/`
`2_0.api.myurl.com/`

This would increase the freedom of the administrator while it reduces complexity in our code - win-win!

Reply:
No that's impossible, because on both sides service-consumer (graphql-client) and service-provider (graphql-endpoint) the correct client and resolver will be selected by the api-version define as enum-value and must match with the correct directory-naming and -structure.
The more variants you will support the more complexity the code will get. So there is no win-win and the admin will not know what is the correct configuration.
We define one exact matching configuration for api-each version and we will only extend the supported variants per as soon as there growup concrete requirements.

Reply:
I disagree on this point - by removing this restriction we would allow greater freedom while we could reduce code on both sides - client and server. The generation of the URL client side is no longer needed and the URL must never be generated server side besides when starting the corresponding services. We should discuss this - I offer to implement this.

Reply:
your arguments are NOT true!!!
there always exists a directory structure for api versions on both sides, because the graphql-schema-generation needs them to select the correct classes. This is part of the typescript code and the evaluated solution to support several api-version on graphql.

there is no need to give more variaties in defining the endpoint urls for cross-community communication. let us spend our efforts in going further with cross-community topics and not loosing time and resource in such 'unimportant' configs.